OpencartTest/src/com/opencart/pages/HomePage.java

```diff
@@ -1,13 +1,32 @@
 package com.opencart.pages;
 
+import com.opencart.pages.search.ProductContainersComponent;
+import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class HomePage extends AbstractPageWithHeader {
 
+    private List<ProductContainersComponent> productContainersComponents;
+
     // CompareButtonMainPage
-   // "//a[text()='MacBook']/../../following-sibling::div/button[@data-original-title='Add to Wish List']"
+    // "//a[text()='MacBook']/../../following-sibling::div/button[@data-original-title='Add to Wish List']"
```

Comment:
what's gonna happen if macbook product is deleted ?

Reply:
it was already commented  - so this trash code i removed